seaborn/_core/data.py

```diff
@@ -258,3 +265,26 @@ def _assign_variables(
         frame = pd.DataFrame(plot_data)
 
         return frame, names, ids
+
+
+def convert_dataframe_to_pandas(data: object) -> pd.DataFrame:
+    """Use the DataFrame exchange protocol, or fail gracefully."""
+    if isinstance(data, pd.DataFrame):
+        return data
+    if not hasattr(pd.api, "interchange"):
+        msg = (
+            "Support for non-pandas DataFrame objects requires a version of pandas"
+            "that implements the DataFrame interchange protocol. Please upgrade "
+            "your pandas version or coerce your data to pandas before passing "
+            "it to seaborn."
+        )
+        raise TypeError(msg)
+
+    try:
+        return pd.api.interchange.from_dataframe(data)
+    except Exception as err:
```

Comment:
I'd strongly suggest setting 2.0.2 as the minimum version to try interchanging from, because there's some pretty basic mistakes in earlier versions:
```python
In [1]: df = pl.DataFrame({'a': [1,2,3]})

In [2]: pd.api.interchange.from_dataframe(df[1:])
Out[2]:
                 a
0                2
1                3
2  125822987010162
```
(😱 )

Reply:
Ah, I didn't realize there were silent failure modes.

Reply:
I do sort of continue to think that it's not seaborn's business to pin behaviors on specific upstream iterations, but I guess "sometimes wrong" is worse than "sometimes errors"...

Reply:
sure, your decision (obvs), just pointing this out because as the conversion happens internally, it wouldn't be obvious to users that the issue is in pandas, and so you'd likely get tonnes of issues in your issue tracker